aws/resource_aws_cloudwatch_dashboard.go

```diff
@@ -0,0 +1,139 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudwatch"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsCloudWatchDashboard() *schema.Resource {
+	return &schema.Resource{
+		Create:        resourceAwsCloudWatchDashboardCreate,
+		Read:          resourceAwsCloudWatchDashboardRead,
+		Update:        resourceAwsCloudWatchDashboardUpdate,
+		Delete:        resourceAwsCloudWatchDashboardDelete,
+		SchemaVersion: 1,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"dashboard_arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"dashboard_body": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateJsonString,
+				StateFunc: func(v interface{}) string {
+					json, _ := normalizeJsonString(v)
+					return json
+				},
+			},
+			"dashboard_name": {
+				Type:     schema.TypeString,
+				Optional: true,
```

Comment:
This should be replaced by `Required: true` as name & body are required by the API